DeepSpeech.ipynb

```diff
@@ -1337,6 +1351,55 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
+    "Finally, we restore the trained variables into a simpler graph that we can export for serving."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "with tf.device('/cpu:0'):\n",
+    "    model_exporter = exporter.Exporter(saver)\n",
+    "    \n",
+    "    # Run inference\n",
+    "    input = tf.placeholder(tf.float32, [None, None, n_input + 2*n_input*n_context])\n",
+    "    \n",
+    "    n_items  = tf.slice(tf.shape(input), [0], [1])\n",
+    "    n_steps = tf.slice(tf.shape(input), [1], [1])\n",
+    "    seq_length = tf.tile(n_steps, n_items)\n",
+    "    logits = BiRNN(input, tf.to_int64(seq_length))\n",
+    "    \n",
+    "    logits = tf.transpose(logits, [1, 0, 2])\n",
+    "    decoded, _ = ctc_ops.ctc_beam_search_decoder(logits, seq_length)\n",
+    "    decoded = tf.sparse_tensor_to_dense(decoded[0])\n",
```

Comment:
What might be nice to do, and does not look like it would be too hard, is to output multiple results. Something like
```python
decoded = [tf.sparse_tensor_to_dense(d) for d in decoded]
```
It's pretty standard for STT engines to do this and it lets downstream code with domain knowledge to disambiguate between the various results.